### PR TITLE
feat(button): add support to small variation

### LINKED ISSUE #4411: Button - add "small" variation
https://github.com/patternfly/patternfly/pull/3132/

Adds a small variation to any button with `.pf-m-small`

### PR DESCRIPTION
**Pull request by boaz0**
**What**: Closes #4411 

//cc @mcoker

### CODE REVIEW

**Comment by boaz0** [comment]
@jenny-s51 I forgot to add an example! :sob: 
working on it - thanks for the helpful feedback.

File: packages/react-core/src/components/Button/examples/Button.md [hunk 1]
```diff
@@ -168,3 +168,15 @@ ButtonTypes = () => (
...
 ```
+
+```js title=Small
+import React from 'react';
+import { Button } from '@patternfly/react-core';
+
+ButtonSmallVariants = () => (
+  <React.Fragment>
+    <Button variant="primary" isSmall>Primary</Button> <Button variant="secondary" isSmall>Secondary</Button>{' '}
+    <Button variant="tertiary" isSmall>Tertiary</Button> <Button variant="danger" isSmall>Danger</Button><br /><br />
```
Comment (mcoker) [inline packages/react-core/src/components/Button/examples/Button.md:179]: Small nit, but could each of these be on their own line? And are the breaks at the end necessary?

File: packages/react-core/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -50,6 +50,8 @@ export interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
   icon?: React.ReactNode | null;
   /** Sets the button tabindex. */
   tabIndex?: number;
+  /** Adds small styling to a link button */
```
Comment (tlabaj) [inline packages/react-core/src/components/Button/Button.tsx:53]: I don't think this styling is specific to link buttons.  Can you update the description here.

File: packages/react-core/src/components/Button/examples/Button.md (hunk 1 above)
Comment (boaz0) [reply packages/react-core/src/components/Button/examples/Button.md:179]: Yes, sure!

File: packages/react-core/src/components/Button/Button.tsx (hunk 2 above)
Comment (boaz0) [reply packages/react-core/src/components/Button/Button.tsx:53]: Sure.

**Comment by boaz0** [comment]
Thank you @jenny-s51 and @tlabaj for the review :bow:

**Pull request merged**